Technics KN5000 musical keyboard

felipesanches opened the pull request:
This is a much smaller pull request with a smaller amount of code, in the hope to facilitate code-reviews. Once this lands, I'll open subsequent PRs with additional chunks, to be reviewed as well.

The full wip patchset is still available at PR #11835, but that seems to be too big a fish to fry all at once...

happppp commented:
I'm 50/50 on the company string being the brand "Technics", but as I mentioned on Discord the other day IMO the driver file belongs in the Matsushita folder.
https://en.wikipedia.org/wiki/Technics_(brand)

felipesanches commented:
I've made the changes suggested by code-reviewers (both here and also via private chat) and then squashed everything into a single commit.

For reference, here were the fixes made after code-reviews:
* some minor tweaks (remove "Keyboard" from the device title, reorder includes)
* properly declared the checking device LEDs
* moved the kn5000 driver to the `src/mame/matsushita` directory
* simplified ioport reads for the control panel buttons

felipesanches commented:
@cuavas, @rb6502 and others, last week I made a few final adjustments based on the initial feedback provided by @happppp and this PR is ready for your code-review & merge.

rb6502 commented:
Thanks @felipesanches!

rb6502 commented on src/mame/matsushita/kn5000.cpp:
```diff
@@ -0,0 +1,845 @@
+} // anonymous namespace
+
+//   YEAR  NAME   PARENT  COMPAT  MACHINE INPUT   STATE         INIT        COMPANY      FULLNAME             FLAGS
+CONS(199?, kn5000,    0,       0, kn5000, kn5000, kn5000_state, empty_init, "Technics", "SX-KN5000", MACHINE_NOT_WORKING|MACHINE_NO_SOUND)
```
Looks like 1998: https://www.soundonsound.com/reviews/technics-kn5000

felipesanches commented:
Thanks, @rb6502!
